docs/reST/ref/event.rst

```diff
@@ -369,6 +369,11 @@ On Android, the following events can be generated
    If ``None`` is passed as the argument, ALL of the event types are blocked
    from being placed on the queue.
 
+   Note: Blocking some kinds of events (like windowevents) could potentially
+   break internal pygame machinery that relies on these events being unblocked,
+   so use this function with caution. When unsure, it is better to simply
+   ignore an event during event handling, rather than trying to block the event.
```

Comment:
This is diffuse and confusing. Why can't this be dealt with internally?

Reply:
Because we don't know if blocking the internal processing of events is intended by the end user or not!

Reply:
How much control should the end user have over the *internal processing* of events? 

Not sure if the end user is aware and expects that some internal pygame machinery breaks or behaves differently when blocking some events (which ones?). I would expect that the only thing that happens when blocking events is that my python code does not receive those events but pygame still behaves 'as usual'.

I might be wrong thought. Since there is a simple workaround (just don't block nor process those events) this fine with me.

Reply:
A quick local test shows that blocking `pygame.KEYDOWN` has no effect on `pygame.key.get_pressed()`. But blocking `WINDOWEVENT` can break some things [here in SDL](https://github.com/libsdl-org/SDL/blob/4b38d4c96b03b6e1e845ee2a0338d78aca652f5a/src/video/cocoa/SDL_cocoametalview.m#L38) as well as in pygame `display.c`.
Problem is that we cannot accurately predict nor document the events whole blocking can affect internal state because it is fairly platform and SDL-version dependent

Reply:
Hm, ok, quick thought: never block WINDOWEVENT  in SDL but if the user blocks it then just filter it out of the events that are returned? I guess the effort to implement this is not worth it.

Reply:
That is pretty much the existing implementation but it means that a custom implementation of `SDL_WaitEventTimeout` has to be used